### PR TITLE
[API_PARSER][CYBEREASON] Fix multiple points

### PR DESCRIPTION
**Pull request by nlancon**
### Fixed

- [API_PARSER][CYBEREASON] Fix multiple points:
    - Add a "to" argument to fetch malwares
    - Add two distinct timestamps for malops & malwares
    - Increment timestamp if no logs where retrieved during the last 24hours

### CODE REVIEW

File: vulture_os/services/frontend/models.py [hunk 1]
```diff
@@ -748,6 +748,9 @@ class Frontend(models.Model):
         default="",
         verbose_name=_("Cybereason password for authentication")
     )
+    cybereason_timestamp = models.JSONField(
```
Comment (frikilax) [inline vulture_os/services/frontend/models.py:751]: there is a `last_collected_timestamps` that already holds all generic timestamps, no need to create a new one
see the WAF_BARRACUDA collector for an example

File: vulture_os/toolkit/api_parser/cybereason/cybereason.py [hunk 2]
```diff
@@ -23,13 +23,15 @@
 __doc__ = 'Cybereason API Parser toolkit'
 __parser__ = 'CYBEREASON'
 
+import sys
```
Comment (frikilax) [inline vulture_os/toolkit/api_parser/cybereason/cybereason.py:26]: unused import

File: vulture_os/toolkit/api_parser/cybereason/cybereason.py [hunk 3]
```diff
@@ -23,13 +23,15 @@
...
+
 from django.conf import settings
 from toolkit.api_parser.api_parser import ApiParser
 from django.utils import timezone
 
 import json
 import logging
-from datetime import timedelta
+from datetime import timedelta, datetime
```
Comment (frikilax) [inline vulture_os/toolkit/api_parser/cybereason/cybereason.py:34]: unused import

File: vulture_os/toolkit/api_parser/cybereason/cybereason.py [hunk 4]
```diff
@@ -546,8 +552,13 @@ def execute(self):
             # Writting may take some while, so refresh token in Redis
             self.update_lock()
 
-        # update last_api_call only if logs are retrieved
-        self.frontend.last_api_call = to
-
+            if len(logs) > 0:
+                # update last_api_call only if logs are retrieved
+                self.frontend.cybereason_timestamp[kind] = to
+            elif self.frontend.cybereason_timestamp.get(kind, timezone.now()) < timezone.now() - timedelta(hours=24):
```
Comment (frikilax) [inline vulture_os/toolkit/api_parser/cybereason/cybereason.py:558]:
```suggestion
            elif since < timezone.now() - timedelta(hours=24):
```

**Review by frikilax — APPROVED** [review]

**Review by ELDiablO59152 — DISMISSED** [review]
Change looks good.

File: vulture_os/toolkit/api_parser/cybereason/cybereason.py [hunk 5]
```diff
@@ -517,18 +520,19 @@ def format_log(self, kind, log):
...
-        # Get last api call or 30 days ago
-        since = self.frontend.last_api_call or (timezone.now() - timedelta(days=30))
-        # 24h max per request
-        to = min(timezone.now(), since + timedelta(hours=24))
 
-        # delay the times of 5 minutes, to let the times at the API to have all logs
-        to = to - timedelta(minutes=5)
+        for kind in ["malops", "malwares"]:
+            # Get last api call or 30 days ago
+            since = self.last_collected_timestamps.get(f"cybereason_{kind}") or (timezone.now() - timedelta(days=30))
```
[the remark this answers is not in the document]
Comment (ELDiablO59152) [reply vulture_os/toolkit/api_parser/cybereason/cybereason.py]: I pushed the change suggested and tested with a new listener

**Pull request merged**